### ISSUE TITLE
Twitter has banned Third Party clients

### ISSUE THREAD

**Issue by Scotty-Trees** [IBBoard/cawbird#495]
https://www.engadget.com/twitter-new-developer-terms-ban-third-party-clients-211247096.html

I assume this means Cawbird and other Twitters clients are now dead in the water? If I’m mistaken please excuse me and my humblest apologies. I just wanted to bring this to attention since it just happened. Feel free to delete this report or make a statement if it’s not relevant. 

If this is the end, I just want to thank you for all your hard work and dedication on creating and maintaining a wonderful little piece of Linux software. If this is indeed the end, perhaps there could be a way to migrate Cawbird from the Twitter API over to the Mastodon API and keep this software going.

**Comment by IBBoard** [comment]
Yep, pretty much. I expected it would come to this when the big clients got closed down, and I've been posting about it on the [Cawbird twitter feed](https://twitter.com/cawbirdclient). I expect that the only reason why we're not closed yet is because we're too small and too far down the list of API keys when sorted by user count for them to notice us.

We can't migrate this Cawbird to Mastodon, but @CodedOre was already working on [NewCaw](https://github.com/CodedOre/NewCaw/) to replace our clunky old code with nicely structured new Vala that would support both Twitter _and_ Mastodon. Now it's just going to support Mastodon!

**Comment by Scotty-Trees** [comment]
Thanks for the quick reply. I just heard about NewCaw earlier this morning thankfully and immediately starred that as a favorite to keep an eye on. That sure is rough you had to deal with having the expectation that at some point your software was going to get nuked, I can't imagine that kind of headache. 

I'll keep an eye following the @cawbird@mastodon.social handle in the meantime. I use Fedora 37 btw, but I assume NewCaw will be released only as a flatpak? No worries on how it's released, just curious. Also, do you know if there is an ETA for NewCaw? I understand if it's more of a "it's ready when it's ready" type of scenario. In any case, here's hoping Mastodon keeps gaining tracking over the coming years!

**Comment by IBBoard** [comment]
Twitter has always screwed third-party client developers over. Between features with no API, killing off streaming, rolling out the useless bits of the v2 API first, and putting in place app-wide request limits, then I'm kind of used to it.

Personally, I _hate_ FlatPak and Snap etc. RPM is the superior packaging format, so I'll make sure that there's an RPM available via [the OBS account](https://build.opensuse.org/project/show/home:IBBoard:cawbird).

And yeah, it's a "ready when it's ready" situation. It's a hobby project. I've got work, CodedOre has university. We do what we can when we can.

**Comment by starvald** [comment]
thought cawbird had been blocked this morning, tweets refused to load but deleted the config file and twitter let me authorize it again.

**Comment by IBBoard** [comment]
Seems it happened to the `@cawbirdclient` account. But not to my account or to the test account that I use for posting junk to check calls without bothering other people 🤔

**Comment by Nardol** [comment]
It looks like over now :disappointed:
I had no tweets displayed when I launched Cawbird, I deleted my config and now Cawbird show an alert simply named "unauthorized" when I click the button to ask for the PIN.

**Comment by IBBoard** [comment]
Ah. Worked out what happened. I've got custom builds that use dev keys (because Twitter allowed you to create three sets of keys per app, to separate Live from Testing). The `@cawbirdclient` account was using the Live keys. My other accounts were on the Dev keys. And when I re-authenticated then Cawbird would have used the Dev keys.

So yeah, the app has been killed off by Twitter now. Unless people are using "Unstable". And even that will go eventually.

**Comment by starvald** [comment]
shame, guess i'll end up going by my standby of using a browser and auto reload addon when mine dies.

**Comment by 20NE** [comment]
Thx for your work!
